larq_compute_engine/core/bgemm_kernels_common.h

```diff
@@ -2,30 +2,36 @@
 #define COMPUTE_EGNINE_TFLITE_KERNELS_BGEMM_KERNELS_COMMON_H_
 
 #include "larq_compute_engine/core/bconv2d_output_transform.h"
-#include "tensorflow/lite/experimental/ruy/kernel_common.h"
+#include "ruy/kernel_common.h"
 
 using namespace ruy;
 
 using compute_engine::core::OutputTransform;
 
-// Our version of `ruy::BasicSpec`
-// Original is in `lite/experimental/ruy/spec.h`
-// We simply use our `OutputTransform` struct
+// Our version of `ruy::MulParams`; The original is in `ruy/mul_params.h`.
+// We simply use our `OutputTransform` struct.
 template <typename tAccumScalar, typename tDstScalar>
-struct BinaryBasicSpec {
+struct BinaryMulParams {
   using AccumScalar = tAccumScalar;
   using DstScalar = tDstScalar;
 
   OutputTransform<AccumScalar, DstScalar> output_transform;
 
-  // This is identical to `ruy::BasicSpec`
+ public:
```

Comment:
`structs` attributes/methods are by default `public`. the `public` identifier here  is not necessary.

Reply:
Makes sense, but just for reference it was copied from here: https://github.com/google/ruy/blob/bec99d61aa9d1e4d07a85f2be326821a1600593c/ruy/mul_params.h#L125-L145

Reply:
wow, they have completely changed the Ruy spec impl. if you look a few lines above now is a `class` and not a `struct`.  We don't need other parts of the new impl or RUY spec/Mul_params?

Reply:
also its not a final class and we can actually subclass it to add our extensions to it without losing the compatibility.

Reply:
@Tombana does it make sense to integrate/extend our `output_transorm` functionality to that?

Reply:
Ah interesting. I don't think we were thinking too much about it at the time, just trying to get things to successfully compile. The reason we didn't include the other stuff in the new `MulParams` class is that we have the output multiplier, zero point, clamp min/max etc in other parts of LCE (e.g. the `OutputTransform` struct) and didn't want to have to mess around with that too.

Reply:
Since some time ago, at the PR where we started using our own `OutputTransform`, I think we basically don't use these specs/mulparams anymore, except for passing through our own outputtransform.

Reply:
Would you rather extend the `MulParams` class?

Reply:
I think me and @arash discussed in a call whether to extend it (I guess you could subclass it in either direction?) or to have it as a member variable. I think the opinions were rather neutral, and then we opted for the member variable. I'm still neutral.

Reply:
Its a good point that if we make `OutputTransform` a subclass of `MulParams` then we don't have to copy any `MulParams` code, so thats easier to keep up to date.
However this won't work for us since `OutputTransform` is also used in places where we don't have RUY.
We could do it the other way around, having `MulParams` as a subclass of `OutputTransform` (I think that was my original implementation when I pushed the PR), but thats almost exactly the same as the current method.

Reply:
For this PR, if everything works and there is no regression I am in favour of continuing with the current code but I think we should take a deep look into the RUY changes and whats the best way to adopt them.